quinn/src/connection.rs

```diff
@@ -277,6 +272,53 @@ impl FuturesStream for IncomingStreams {
     }
 }
 
+/// A future that will resolve into an opened outgoing unidirectionnal stream
+pub struct OpenUni {
+    recv: oneshot::Receiver<Result<(StreamId, bool), ConnectionError>>,
+    conn: ConnectionRef,
+}
+
+impl Future for OpenUni {
+    type Item = SendStream;
+    type Error = ConnectionError;
+
+    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        match self.recv.poll() {
+            Err(_) => unreachable!(),
```

Comment:
```suggestion
            Err(_) => unreachable!("oneshot sender won't be dropped while `self.conn` is keeping the `ConnectionInner` alive"),
```
And similar for `OpenBi`.